### PR TITLE
Add CustomControl components

### LINKED ISSUE #6: Custom control buttons

### PR DESCRIPTION
**Pull request by Kanahiro**
Close #6 

### What I did

- add `CustomControl` components to add user-defined Control.
- you can find example in http://localhost:5173/examples/custom-control

### Notes

None

## Release Notes

- **New Features**
  - Introduced a new "Custom Control" component for enhanced map interactions, allowing users to toggle terrain and hillshade layers.
  - Added a link to the "Custom Control" example in the examples list.

- **Documentation**
  - Created a new markdown file detailing the "Custom Control" component, including usage instructions and code snippets.

- **Version Update**
  - Incremented project version from 0.0.5 to 0.0.6.

### CODE REVIEW

File: src/lib/maplibre/controls/CustomControl/Icon.svelte [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<script lang="ts">
+	import type { Snippet } from 'svelte';
+	let {
+		children,
+		onclick,
+		oncontextmenu,
+		className
+	}: { children?: Snippet; className: string; onclick?: () => void; oncontextmenu?: () => void } = $props();
```
Comment (ciscorn) [inline src/lib/maplibre/controls/CustomControl/Icon.svelte:8]: I'd recommend importing `MouseEventHandler` from `'svelte/elements'` and using it as shown below:

```suggestion
		onclick?: MouseEventHandler<HTMLButtonElement>;
		oncontextmenu?: MouseEventHandler<HTMLButtonElement>;
	} = $props();
```

File: src/lib/maplibre/controls/CustomControl/Icon.svelte [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<script lang="ts">
+	import type { Snippet } from 'svelte';
+	let {
+		children,
+		onclick,
+		oncontextmenu,
+		className
```
Comment (ciscorn) [inline src/lib/maplibre/controls/CustomControl/Icon.svelte:7]: This is an idiom I frequently come across!

```suggestion
		class: className
```

File: src/content/examples/custom-control/CustomControl.svelte [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<script lang="ts">
+	import { HillshadeLayer, MapLibre, RasterDEMTileSource, Terrain, CustomControl } from 'svelte-maplibre-gl';
+	import IconMoon from './IconMoon.svelte';
+	import IconSun from './IconSun.svelte';
```
Comment (ciscorn) [inline src/content/examples/custom-control/CustomControl.svelte:4]: In our project setup, we could also import these icons from `lucide-svelte`, but your approach is perfectly fine.

```
	import Sun from 'lucide-svelte/icons/sun';
	import Moon from 'lucide-svelte/icons/moon';
```

**Review by ciscorn — CHANGES_REQUESTED** [review]
Great!
I’ve made a few comments suggesting improvements.

**Comment by Kanahiro** [comment]
@ciscorn Thank you for comments! I revised as you advised.

**Review by ciscorn — APPROVED** [review]
@Kanahiro I've made some simplifications!

- Removed `CustomControl.Group` as it can now be represented as `<CustomControl group={true}>`
- Also removed `CustomControl.Group` since it can be expressed with a standard HTML `<button>`.  

Let me know if you have any thoughts or suggestions!

**Pull request merged**